### PR TITLE
pagination is added to the get method

### LINKED ISSUE #16: Introduce pagination for the get all todos route
Use query parameters to paginate the get all todos route. 

- limit
- offset
- sort 
- order(set to default by ascending order based creation time)

### PR DESCRIPTION
**Pull request by msm-faraj**
pagination capability added to the get method with the initial value of three(just for test) but it could reset to anything in the future.

### CODE REVIEW

File: restfull-api/controllers/todo.js [hunk 1]
```diff
@@ -5,7 +5,16 @@ class TodoController {
   }
 
   async getAllTodos(req, res) {
-    const todos = await this.Todo.find();
+    const page = req.query.page || 0;
+    const todoPerPage = 3;
+    const isDoneQuery = { isDone: false };
+    const notDeletedQuery = { deletedAt: null };
+    const isVisible = isDoneQuery && notDeletedQuery;
```
Comment (pouyapourbaba) [inline restfull-api/controllers/todo.js]: Can you explain what you want to achieve here?

File: restfull-api/controllers/todo.js [hunk 2]
```diff
@@ -5,7 +5,16 @@ class TodoController {
   }
 
   async getAllTodos(req, res) {
-    const todos = await this.Todo.find();
+    const page = req.query.page || 0;
+    const todoPerPage = 3;
```
Comment (pouyapourbaba) [inline restfull-api/controllers/todo.js]: This one can come from the query params also.

File: restfull-api/controllers/todo.js (hunk 1 above)
Comment (msm-faraj) [reply restfull-api/controllers/todo.js]: line 9: I assume that in each page, app going to show 3 last todo list
line 10:  is done  todos not going to be shown 
line 11: deleted todos not going to be shown
line 12: logical &

Comment (pouyapourbaba) [reply restfull-api/controllers/todo.js]: log line 12 and check the result

**Review by pouyapourbaba — APPROVED** [review]

**Pull request merged**